odm/abstract_repository.go

```diff
@@ -12,6 +12,19 @@ import (
 	"go.uber.org/zap"
 )
 
+type AbstractRepositoryInterface[T any] interface {
```

Comment:
Let's follow constructs of go-grpc.

Let this interface be named as BootRepository and AbstractRepository struct (line 15) be named as UnimplementedBootRepository.

Thanks

Reply:
I have done the changes of renaming variables, do review it and tell me if i have to do any more changes